singlePost.php

```diff
@@ -19,7 +27,25 @@
     header('Location: index.php');
 }
 
-$formattedDate = date('d/m/Y', strtotime($singlePostDetails->dateTime));
+
+if (isset($_POST['submit']))
+{
+
+    $inputtedComment = strip_tags($_POST['comment']);
+    $currentUserId = $_SESSION['userid'];
+    $currentPostId = $_GET['id'];
+
+    $db = connectToDb();
+    $commentModel = new CommentModel($db);
+
+    if(strlen($inputtedComment) < 10 || strlen($inputtedComment) > 200) {
+        $errorMessage = 'Comment must be between 10 to 200 characters';
+    } else {
+        $commentModel->addComment($inputtedComment, $currentUserId, $currentPostId);
+        $successMessage = 'Congratulations! Your comment has been added to this blog post! Thanks for contributing to our network!';
+        $inputtedComment = '';
+    }
+}
```

Comment:
If you move this code to before you do getSinglePostById - you'll find that the new comment appears as soon as the form is submitted